core/src/test/scala/io/aiven/guardian/kafka/Generators.scala

```diff
@@ -113,6 +114,24 @@ object Generators {
     Gen.choose[Long](head.timestamp, last.timestamp - 1).map(millis => FiniteDuration(millis, MILLISECONDS))
   }
 
+  @SuppressWarnings(
+    Array(
+      "scalafix:DisableSyntax.while"
+    )
+  )
+  private def buildableOfCollCond[C <: Iterable[T], T](cond: C => Boolean, g: Gen[C])(implicit
+      evb: Buildable[T, C]
+  ): Gen[C] =
+    Gen.infiniteLazyList(g).map { ll =>
+      val it   = ll.iterator
+      val bldr = evb.builder
+      while (!cond(bldr.result()))
```

Comment:
Thanks for recommendation, I wanted to avoid this because it involved having to circumvent Scala's lint warnings but I managed to find a way. Just pushed it now to see if tests pass.

Reply:
So I tried a solution, i.e. 

```scala
@SuppressWarnings(
  Array(
    "scalafix:DisableSyntax.while",
    "scalafix:DisableSyntax.return"
  )
)
@scala.annotation.nowarn("msg=return statement uses an exception")
private def buildableOfCollCond[C <: Iterable[T], T](cond: C => Boolean, g: Gen[C])(implicit
    evb: Buildable[T, C]
): Gen[C] =
  Gen.infiniteLazyList(g).map { ll =>
    val it   = ll.iterator
    val bldr = evb.builder
    while (true) {
      val result = bldr.result()
      if (cond(result)) {
        return result
      }
      bldr ++= it.next()
    }
    return bldr.result()
  }
  ```
And it didn't work/terminate. Don't have time to look at this now so I will make an issue to look at it later as an improvement.